### PR TITLE
feat(Table): Added isNoInset prop for flat tree tables

### PR DESCRIPTION
**Pull request by nicolethoen**
Closes https://github.com/patternfly/patternfly-react/issues/8604

### CODE REVIEW

File: packages/react-table/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -57,6 +57,8 @@ export interface TableProps extends React.HTMLProps<HTMLTableElement>, OUIAProps
   isStriped?: boolean;
   /** Flag indicating this table contains expandable rows to maintain proper striping */
   isExpandable?: boolean;
+  /** Flag rows will not have the inset typically reserved for expanding/collapsing rows in a tree table. Intended for use on tree tables with no visible rows with children. */
```
Comment (kmcfaul) [inline packages/react-table/src/components/Table/Table.tsx:60]: `indicating` missing

File: packages/react-table/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -57,6 +57,8 @@ export interface TableProps extends React.HTMLProps<HTMLTableElement>, OUIAProps
...
+  isNoInset?: boolean;
```
Comment (tlabaj) [inline packages/react-table/src/components/Table/Table.tsx:61]: maybe `hasNoInset`

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by tlabaj — APPROVED** [review]

**Review by thatblindgeye — APPROVED** [review]

**Review by mattnolting — APPROVED** [review]
LGTM 👍 Might add a follow up to showcase feature in an example.

**Review by kmcfaul — APPROVED** [review]
lgtm just needs a few lint errors resolved

**Pull request merged**